src/pedal/split.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{pedal, util, Pedal, Result};
+use anyhow::anyhow;
+
+pub struct Split {
+    pedals: Vec<pedal::Boxed>,
+}
+
+impl Split {
+    pub fn new(pedals: Vec<pedal::Boxed>) -> Result<Self> {
+        if pedals.is_empty() {
+            Err(anyhow!("Must have at least one pedal in the Split"))
+        } else {
+            Ok(Self { pedals })
+        }
+    }
+}
+
+impl Pedal for Split {
+    fn name(&self) -> String {
+        "Split".into()
+    }
+
+    fn process(&mut self, input: &[f32], output: &mut [f32]) -> Result<()> {
+        let length = output.len();
+        util::zero_slice(output);
+
+        let mut pedal_output: Vec<f32> = vec![0.0; length];
+
+        for pedal in &mut self.pedals {
+            pedal.process(input, &mut pedal_output)?;
+
+            for i in 0..output.len() {
+                output[i] += pedal_output[i];
```

Comment:
Yeah, that is an issue, and I'm not totally sure how that is normally resolved in practice. You can't just alter the signal as you see it approaching the max, because that would change the perceived frequencies. Doing this generically requires a little latency to build up a buffer, I think, but that wouldn't be acceptable for pedals. Maybe the player is just responsible for the gain on  their input?